src/main/java/net/sf/mzmine/modules/tools/isotopepatternpreview/IsotopePatternPreviewDialog.java

```diff
@@ -0,0 +1,332 @@
+    updateChart(pattern);
+  }
+
+  private void updateChart(ExtendedIsotopePattern pattern) {
+    dataset = new ExtendedIsotopePatternDataSet(pattern, minIntensity, mergeWidth);
+    if (pol == PolarityType.NEUTRAL)
+      chart = ChartFactory.createXYBarChart("Isotope pattern preview", "Exact mass / Da", false,
+          "Abundance", dataset);
+    else
+      chart = ChartFactory.createXYBarChart("Isotope pattern preview", "m/z", false, "Abundance",
+          dataset);
+
+    theme.apply(chart);
+    plot = chart.getXYPlot();
+    plot.addRangeMarker(new ValueMarker(minIntensity, belowMin, new BasicStroke(1.0f)));
+    XYItemRenderer r = plot.getRenderer();
+    r.setSeriesPaint(0, aboveMin);
+    r.setSeriesPaint(1, belowMin);
+    r.setDefaultToolTipGenerator(ttGen);
+    pnlChart.setChart(chart);
+  }
+
+  private boolean updateParameters() {
+    updateParameterSetFromComponents();
+    if (!checkParameters()) {
+      logger.info("updateParameters() failed due to invalid input.");
+      return false;
+    }
+
+    molecule = pMolecule.getValue();
+    if (pCustom.getValue()) {
+      minAbundance = pMinAbundance.getValue();
+      mergeWidth = pMergeWidth.getValue();
+      minIntensity = pMinIntensity.getValue();
+      charge = pCharge.getValue();
+    } else {
+      minAbundance = 0.01;
+      mergeWidth = 0.0005;
+      minIntensity = 0.05;
+      charge = 1;
+    }
+
+    if (charge > 0) {
+      pol = PolarityType.POSITIVE;
+    } else if (charge < 0) {
+      pol = PolarityType.NEGATIVE;
+      charge *= -1;
+    } else {
+      pol = PolarityType.NEUTRAL;
+    }
+
+    return true;
+  }
+
+  private boolean checkParameters() {
+    if (/* pElement.getValue().equals("") */pMolecule.getValue() == null
+        || pMolecule.getValue().equals("")) {
+      logger.info("Invalid input or Element == \"\" and no autoCarbon");
+      return false;
+    }
+    if (pMinAbundance.getValue() == null || pMinAbundance.getValue() > 1.0d
+        || pMinAbundance.getValue() <= 0.0d) {
+      logger.info("Minimun abundance invalid. " + pMinAbundance.getValue());
+      return false;
+    }
+    if (pMinIntensity.getValue() == null || pMinIntensity.getValue() > 1.0d
+        || pMinIntensity.getValue() < 0.0d) {
+      logger.info("Minimum intensity invalid. " + pMinIntensity.getValue());
+      return false;
+    }
+    if (pMergeWidth.getValue() == null || pMergeWidth.getValue() < 0.0d) {
+      logger.info("Merge width invalid. " + pMergeWidth.getValue());
+      return false;
+    }
+
+    logger.info("Parameters valid");
+    return true;
+  }
+
+  private ExtendedIsotopePattern calculateIsotopePattern() {
+    ExtendedIsotopePattern pattern = new ExtendedIsotopePattern();
+
+    if (!checkParameters())
+      return null;
+
+    if (molecule.equals(""))
+      return null;
+
+    logger.info("Calculating isotope pattern: " + molecule);
+
+    try {
+      pattern.setUpFromFormula(molecule, minAbundance, mergeWidth, minIntensity);

```

Comment:
I don't want to have two classes in MZmine that do exactly the same thing - why don't you reuse the existing calculateIsotopePattern() function in  net.sf.mzmine.modules.peaklistmethods.isotopes.isotopeprediction.IsotopePatternCalculator?

Reply:
Because it does not yield a description of the isotope composition, which i needed for IPS and this. It keeps track of the isotope composition, while it's calculating the isotope pattern.
Since i changed a bit of the data structure (having a list instead of an array while calculating) I did not want to break other modules.

Reply:
Sounds reasonable. I will do that when im coming back from vacation (next wednesday), because i think it might take a few hours.

Reply:
In fact, it would be best to move that code to MSDK: 
https://github.com/msdk/msdk/blob/master/msdk-isotopes-isotopepattern/src/main/java/io/github/msdk/isotopes/isotopepattern/IsotopePatternGeneratorAlgorithm.java
Then we can remove the code from MZmine completely and just use the MSDK function.

Reply:
I am going to merge this PR now, and we can deal with these changes later.

Reply:
I will look into it.